pybikes/ciclosampa.py

```diff
@@ -28,31 +27,30 @@ def update(self, scraper=None):
             scraper = utils.PyBikesScraper()
         scraper.setUserAgent(USERAGENT)
 
-        html_data = scraper.request(self.feed_url)
-        stations = re.findall(STATIONS_RGX, html_data)
+        body = json.dumps({"tipo": "getstations"})
+        json_data = scraper.request(
+                self.feed_url,
+                method='POST',
+                data=body,
+                headers={"Content-Type": "application/json"})
+        stations = json.loads(json_data)
 
         self.stations = []
 
         for station in stations:
-            station = station.replace('"', '').split(',')
             self.stations.append(CicloSampaStation(station))
 
 
 class CicloSampaStation(BikeShareStation):
     def __init__(self, data):
-        '''
-        data is a list of strings, in the following order:
-            [latitude, longitude, stationId, name, address, availableBikes,
-             bike capacity]
-        '''
         super(CicloSampaStation, self).__init__()
-        self.name = data[3]
-        self.latitude = float(data[0])
-        self.longitude = float(data[1])
-        self.bikes = int(data[5])
-        self.free = int(data[6])
+        self.name = data['nome']
+        self.latitude = float(data['latitude'])
+        self.longitude = float(data['longitude'])
+        self.bikes = int(data['bikes'])
+        self.free = int(data['vagas'])
         self.extra = {
-            'address': data[4],
-            'uid': int(data[2]),
-            'slots': int(data[6]) + int(data[5])
+            'address': data['endereco'],
+            'uid': int(data['id']),
+            'slots': int(data['bikes']) + int(data['bikes'])
```

Comment:
The `slots` field should reflect the absolute number of spaces on an station. It's common to not be there (also adding slots + bikes is not exact too), so you can remove it altogether, since this system does not provide it.

Reply:
I see the API response also includes a field `status`. Let's add that like
```
extra["status"] = "online" if data['status'] == "Em operação" else "offline"
```
or even, since we do not know which statuses the system has, at least we can assure that we know about "online" like:
```
extra["online"] = data['status'] == "Em operação"
```

Since most API clients support either one or the other.